src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -347,5 +347,148 @@ private static bool IsMultidimensionalInitializer(ImmutableArray<BoundExpression
 
             return inits.Length != 0 && inits[0].Kind == BoundKind.ArrayInitialization;
         }
+
+        private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExpression wrappedExprerssion, bool used, bool inPlace)
+        {
+            ImmutableArray<byte> data = default;
+            int elementCount = -1;
+            TypeSymbol elementType = null;
+
+            if (!_module.SupportsPrivateImplClass)
+            {
+                return false;
+            }
+
+            var ctor = ((MethodSymbol)this._module.Compilation.GetWellKnownTypeMember(WellKnownMember.System_ReadOnlySpan_T__ctor));
+            if (ctor == null)
+            {
+                return false;
+            }
+
+            if (wrappedExprerssion is BoundArrayCreation ac)
```

Comment:
It looks like we're mostly using guard expressions to bail out early, but this is the inverse of that. If you instead did

```c#
if (!(wrappedExpression is BoundArrayCreation ac))
{
    return false;
````
that would be consistent. This also means you can move the `elementCount` check next to `TryGetRawDataForArrayInit`, where the cause and effect will be clearer.

Reply:
A more complete version of this would also have an `else if` that checks for `string` conversions. Then it makes more sense to confine handling of each case to its own block.

I actually see `if (o is T1 t)` pattern used quite often especially if thee are more `else if` alternatives. 

I do not see the proposed shape all that much better than the current. I think we are ok regardless so I will keep it as-is.

---
In reply to: [171990479](https://github.com/dotnet/roslyn/pull/24621#discussion_r171990479) [](ancestors = 171990479)